### PR TITLE
chore: housekeeping 1 - fix lint stuff

### LINKED ISSUE #83: Docs callouts regarding Vercel URL
VERCEL_URL env variable is hardcoded in the library right now. Until this is addressed, we should add a mention in the FAQ for people trying to use uploadthing on things other than Vercel

https://discord.com/channels/966627436387266600/1109774435407376404/1109992619519520829

image

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #83 and also performs some general hoursekeeping to silence some lint rules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,5 +26,5 @@
     "turbo": "^1.9.3",
     "typescript": "5.1.0-beta"
   },
-  "packageManager": "pnpm@7.27.0"
+  "packageManager": "pnpm@8.5.1"
```
Comment (juliusmarminge) [inline package.json:29]: should prevent a bunch of big diffs in the lockfile keeping this sync - we were already using lockfile version 6 so should probably use pnpm 8 too

File: packages/uploadthing/src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import type { PackageJson } from "type-fest";
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const packageJson = require("../package.json") as PackageJson;
+if (!packageJson.version) throw new Error("no version found in package.json");
+export const UPLOADTHING_VERSION = packageJson.version;
```
Comment (markflorkowski) [inline packages/uploadthing/src/constants.ts:6]: Thank you for this, I have been saying for a bit that we should really have a centralized constants file, but had not gotten around to actually doing it.

File: docs/src/pages/index.mdx [hunk 3]
```diff
@@ -39,8 +39,15 @@ import { Tab, Tabs } from "nextra-theme-docs";
 
 ```bash copy
 UPLOADTHING_SECRET=... # A secret key for your app (starts with sk_live_)
+UPLOADTHING_URL=https://yourapp.com # The URL to your app
```
Comment (markflorkowski) [inline docs/src/pages/index.mdx:42]: I almost want to not include this and have the callout say that you can specify `UPLOADTHING_URL` if you are not deploying on Vercel. Also this block probably doesn't need the copy button.

Comment (juliusmarminge) [reply docs/src/pages/index.mdx:42]: Agree!

**Review by t3dotgg — APPROVED** [review]

**Review by markflorkowski — APPROVED** [review]

**Pull request merged**